compiler/ETA/Utils/JAR.hs

```diff
@@ -282,9 +287,19 @@ mergeClassesAndJars jarLocation compress fileAndContents jarSelectors = do
   p <- makeAbsoluteFilePath jarLocation
   copyFile copy p
   withArchive p $ do
+    existingEntries <- getEntries
+    let invalidEntries = filterWithKey (\k _ -> invalidEntrySelector k)  existingEntries
+    mapM_ deleteEntry (keys invalidEntries)
```

Comment:
@rahulmutt Oh yeah. This is the actual fix. The problem is that the copied file initially has those files. If you see the link `copyFile copy p` will copy the jar file to a new location. And then we are acting on that jar file with the function `withArchive`.